learn-data-structures-by-building-a-shopping-cart/index17b.js

```diff
@@ -0,0 +1,67 @@
+const products = [
+  {
+    id: 0,
+    name: "Vanilla buttercream cupcake",
+    price: 2.99
+  },
+  {
+    id: 1,
+    name: "French Macaroon",
+    price: 3.99
+  },
+  {
+    id: 2,
+    name: "Fruit sprinkles cupcake",
+    price: 3.99
+  },
+  {
+    id: 3,
+    name: "Pink flower cupcake",
+    price: 5.99
+  }
+];
+
+class ShoppingCart {
+  constructor() {
+    this.items = [];
+    this.discountPercentage = 30;
+  }
+
+  addItem(id, products) {
+    const item = products.find(item => item.id === id);
+    this.items.push(item);
+  }
+
+  getItems() {
+    return this.items;
+  }
+  
+  getCount() {
+    return this.items.length;
+  }
+  
+  removeItem(id) {
+    const index = this.items.indexOf(id);
+    this.items.splice(index, 1);
+  }
+  
+  clearCart() {
+    this.items = [];
+  }
+  
+  applyDiscount(amount) {
+    return (this.discountPercentage / 100 * amount).toFixed(2);
+  }
+}
+
+/*
+You will now see that the number has been rounded and its type is string!
```

Comment:
As you can see, the number has been rounded to two decimal places and its type is `"string"`!
We still want to work with numbers, so use the built-in `parseFloat()` method to convert the string back to a number.
For example: 
```
console.log(parseFloat((1 / 3).toFixed(2))); // 0.33
```